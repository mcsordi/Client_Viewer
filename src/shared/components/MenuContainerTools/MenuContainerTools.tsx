import React from 'react';

type TContainerTools = {
  visible?: boolean;
  children: React.ReactNode;
};
export const MenuContainerTools: React.FC<TContainerTools> = ({
  visible,
  children,
}) => {
  return (
    visible && (
      <div className="flex gap-3 sm:gap-2 md:gap-3 xl:gap-5 flex-wrap">
        {children}
      </div>
    )
  );
};
