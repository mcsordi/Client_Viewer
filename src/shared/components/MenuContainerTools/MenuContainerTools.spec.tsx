import { render, screen } from '@testing-library/react';
import { MenuContainerTools } from './MenuContainerTools';
import React from 'react';

describe('<MenuContainerTools/>', () => {
  it('should render a MenuContainerTools component', () => {
    render(
      <MenuContainerTools visible>
        <p>teste</p>
      </MenuContainerTools>,
    );
    const textContainer = screen.getByText('teste');
    expect(textContainer.parentElement).toBeInTheDocument();
  });
  it('should render a MenuContainerTools component and not be rendering in the document', () => {
    const { container } = render(
      <MenuContainerTools visible={false}>
        <p>teste</p>
      </MenuContainerTools>,
    );

    expect(container).not.toHaveClass();
    expect(container.children).toHaveLength(0);
  });
  it('should match a snapshot', () => {
    const { container } = render(
      <MenuContainerTools visible>
        <p>teste</p>
      </MenuContainerTools>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
