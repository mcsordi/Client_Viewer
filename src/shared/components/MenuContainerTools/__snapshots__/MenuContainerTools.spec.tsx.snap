// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuContainerTools/> should match a snapshot 1`] = `
<div
  class="flex gap-3 sm:gap-2 md:gap-3 xl:gap-5 flex-wrap"
>
  <p>
    teste
  </p>
</div>
`;
